test/byContract/IexecEscrow/IexecEscrowNative.ts

```diff
@@ -0,0 +1,244 @@
+// SPDX-FileCopyrightText: 2020-2024 IEXEC BLOCKCHAIN TECH <contact@iex.ec>
+// SPDX-License-Identifier: Apache-2.0
+
+import { AddressZero } from '@ethersproject/constants';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BigNumber } from 'ethers';
+import { ethers, expect } from 'hardhat';
+import CONFIG from '../../../config/config.json';
+import { loadHardhatFixtureDeployment } from '../../../scripts/hardhat-fixture-deployer';
+import { IexecInterfaceNative, IexecInterfaceNative__factory } from '../../../typechain';
+import { getIexecAccounts } from '../../../utils/poco-tools';
+
+const depositAmount = BigNumber.from(100);
+const etherDepositAmount = ethers.utils.parseUnits(depositAmount.toString(), 9);
+const depositArgs = [{ value: etherDepositAmount }] as [{ value: BigNumber }];
+
+// TODO: remove this when poco is also available in Native mode
+if (CONFIG.chains.default.asset === 'Native') {
+    describe('EscrowNative', () => {
+        let proxyAddress: string;
+        let [iexecPoco, iexecPocoAsAnyone]: IexecInterfaceNative[] = [];
+        let [iexecAdmin, anyoneA, anyoneB, anyone]: SignerWithAddress[] = [];
+
+        beforeEach('Deploy', async () => {
+            proxyAddress = await loadHardhatFixtureDeployment();
+            await loadFixture(initFixture);
+        });
+
+        async function initFixture() {
+            const accounts = await getIexecAccounts();
+            ({ iexecAdmin, anyone: anyoneA, requester: anyoneB, anyone } = accounts);
+
+            iexecPoco = IexecInterfaceNative__factory.connect(proxyAddress, iexecAdmin);
+            iexecPocoAsAnyone = iexecPoco.connect(anyone);
+        }
+
+        describe('Receive and Fallback', () => {
+            it('Should call receive successfully', async () => {
+                await expect(
+                    await anyoneA.sendTransaction({
+                        to: iexecPoco.address,
+                        value: etherDepositAmount,
+                    }),
+                )
+                    .to.changeEtherBalances(
+                        [anyoneA, iexecPoco],
+                        [-etherDepositAmount, etherDepositAmount],
+                    )
+                    .to.changeTokenBalances(iexecPoco, [anyoneA], [depositAmount])
+                    .to.emit(iexecPocoAsAnyone, 'Transfer')
+                    .withArgs(AddressZero, anyoneA.address, depositAmount);
+            });
+
+            it('Should call fallback successfully', async () => {
+                const randomData = ethers.utils.hexlify(
+                    ethers.utils.toUtf8Bytes((Math.random() * 0xfffff).toString(16)),
+                );
+
+                await expect(
+                    await anyoneA.sendTransaction({
```

Comment:
```suggestion
                await expect(
                    anyoneA.sendTransaction({
```
You can remove double "await stage" like that everywhere